theano/tensor/nnet/corr_gemm.c

```diff
@@ -203,15 +208,20 @@ PyArrayObject* corrMM(PyArrayObject* bottom,
         output = top;
         // valid correlation: im2col, then gemm
         // Iterate over batch
+        #pragma omp parallel for schedule(static)
         for (int n = 0; n < batchSize; n++) {
+            int colidx = 0;
+#if defined(_OPENMP)
+            colidx = omp_get_thread_num();
+#endif
             // First, im2col
             im2col((%(float_type)s*)PyArray_DATA(bottom) + n * bottom_stride, nChannels, bottomHeight,
-                    bottomWidth, kH, kW, padH, padW, dH, dW, (%(float_type)s*)PyArray_DATA(col));
+                    bottomWidth, kH, kW, padH, padW, dH, dW, (%(float_type)s*)PyArray_GETPTR3(col, colidx, 0, 0));
```

Comment:
Could it be the PyArray_DATA or PyArray_GETPTR3 that cause this problem? Can this be moved outside the loop for _DATA and for getptr3, could this be computated manually? This is the only python code in the parallel loop, so I don't see what else could cause the double free related to python in the error message.